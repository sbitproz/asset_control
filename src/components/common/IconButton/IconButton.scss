@import "../../../styles/abstracts/variables";

@keyframes fadeIn {
  0% {
    background-color: $color-white;
  }
  100% {
    background-color: $color-grey-lightest;
  }
}

$icon-size: 2rem;

.icon-button {
  width: $icon-size;
  height: $icon-size;
  cursor: pointer;
  display: flex;
  justify-content: center;
  align-items: center;

  &--disabled {
    opacity: $diabled-opacity;
    pointer-events: none;
    cursor: default;
  }

  &:hover {
    background-color: $color-grey-lightest;
    animation: fadeIn ease 0.5s;
  }
}
