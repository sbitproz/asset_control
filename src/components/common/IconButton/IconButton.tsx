import React, { memo } from "react";

// Styles
import "./IconButton.scss";

interface IconButtonProps {
  testid?: string;
  disabled?: boolean;
  onClick: () => void;
  children: React.ReactNode | React.ReactNodeArray;
  className?: string;
}

const IconButton: React.SFC<IconButtonProps> = ({
  testid = "",
  disabled,
  children,
  onClick,
  className = "",
}) => {
  return (
    <div
      data-testid={testid}
      className={`icon-button ${
        disabled ? "icon-button--disabled" : ""
      } ${className}`}
      onClick={onClick}
    >
      {children}
    </div>
  );
};

export default memo(IconButton);
