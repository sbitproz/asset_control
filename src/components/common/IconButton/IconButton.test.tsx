import React from "react";
import { shallow } from "enzyme";

// Own
import IconButton from "./IconButton";
import { testSelector } from "../../../tests/testHelper";

describe("PaginationNav", () => {
  let wrapper: any;

  const className = "heading-class";
  const testId = "icon-button-test-id";
  const disabledClass = "icon-button--disabled";
  const childrenContent = "contents";

  const handleClicked = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <IconButton
        testid={testId}
        disabled={true}
        className={className}
        onClick={handleClicked}
      >
        <div>{childrenContent}</div>
      </IconButton>
    );
  });

  it("should match snapshot", () => {
    expect(wrapper).toMatchSnapshot();
  });

  it("should find heading with testid", () => {
    expect(wrapper.find(testSelector(testId)).exists()).toBe(true);
  });

  it(`should have div with class ${className}`, () => {
    expect(wrapper.find(`div${testSelector(testId)}`).hasClass(className)).toBe(
      true
    );
  });

  it(`shouldn't have div with class ${className}`, () => {
    wrapper.setProps({ className: undefined });
    expect(wrapper.find(testSelector(testId)).hasClass(className)).not.toBe(
      true
    );
  });

  it(`should show a disabled class when disabled`, () => {
    expect(wrapper.find(testSelector(testId)).hasClass(disabledClass)).toBe(
      true
    );
  });

  it(`shouldn't show a disabled class when not disabled`, () => {
    wrapper.setProps({ disabled: false });
    expect(wrapper.find(testSelector(testId)).hasClass(disabledClass)).not.toBe(
      true
    );
  });

  it(`should contain ${childrenContent}`, () => {
    wrapper.setProps({ disabled: false });
    expect(wrapper.find(testSelector(testId)).text()).toContain(
      childrenContent
    );
  });

  it(`should call function when button clicked`, () => {
    expect(wrapper.find(testSelector(testId)).simulate("click"));
    expect(handleClicked).toHaveBeenCalled();
  });
});
