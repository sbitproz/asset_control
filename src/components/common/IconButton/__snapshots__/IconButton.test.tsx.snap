// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaginationNav should match snapshot 1`] = `
<div
  className="icon-button icon-button--disabled heading-class"
  data-testid="icon-button-test-id"
  onClick={[MockFunction]}
>
  <div>
    contents
  </div>
</div>
`;
